### ISSUE TITLE
DigitalTwinsClient service_client.list_models() 

### ISSUE THREAD

**Issue by BillmanH** [Azure/azure-sdk-for-python#15327]
** TLDR if it is supposed to return a list of _ALL_ models, why does it require a `model_id`?**

- **Package Name**: azure-digitaltwins-core
- **Package Version**: 1.0.0
- **Operating System**: Windows
- **Python Version**: 3.7.9

**Describe the bug**
[readme for digital twins is misleading](https://github.com/Azure/azure-sdk-for-python/tree/4559e19e2f3146a49f1eba1706bb798071f4a1f5/sdk/digitaltwins/azure-digitaltwins-core), the doc says that `service_client.list_models()` requires `model_id`, which it shouldn't because it is the function that lists the models. Looking at the actual code for that function shows that `model_id` is not required. Leaving it out causes a crashed in the wrapper function `list_models() missing 1 required positional argument: 'dependencies_for'`  this is at `core\tracing\decorator.py`.   What is the `dependencies for` and how do I get one? Could you please add it to the docs, or remove it if it is not needed? I assume that, because the `list_models()` function _lists the models_ that it doesn't require you to enter a specific modle. 

the error is:
`TypeError: list_models() missing 1 required positional argument: 'dependencies_for'`

Full traceback:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-0822bd9dc56c> in <module>
----> 1 serviceClient.list_models()

~\Anaconda3\envs\iot\lib\site-packages\azure\core\tracing\decorator.py in wrapper_use_tracer(*args, **kwargs)
     81             span_impl_type = settings.tracing_implementation()
     82             if span_impl_type is None:
---> 83                 return func(*args, **kwargs)
     84 
     85             # Merge span is parameter is set, but only if no explicit parent are passed
```

**To Reproduce**
```python
from azure.identity import DefaultAzureCredential
from azure.digitaltwins.core import DigitalTwinsClient

credential = DefaultAzureCredential()
serviceClient = DigitalTwinsClient("home-test-twin.api.wcus.digitaltwins.azure.net", credential)
serviceClient.list_models()
```

**Expected behavior**
not crash, return a list of models. 

**Screenshots**

**Additional context**

**Comment by annatisch** [comment]
Thanks @BillmanH! We are aware of this issue and have a fix pending here:
https://github.com/Azure/azure-sdk-for-python/pull/15157

**Comment by BillmanH** [comment]
Looks like [this fix](https://github.com/Azure/azure-sdk-for-python/pull/15157) did resolve this issue. 

Correction on my earlier comment: `serviceClient.list_models()` returns a generator, not a list.

**Comment by annatisch** [comment]
Thanks for confirmation @BillmanH!
